Makefile

```diff
@@ -9,8 +9,10 @@ pydep:
 .PHONY: pydep-upgrade
 pydep-upgrade:
 	pip install -U pip-tools
-	pip-compile --output-file=pymoose/requirements-dev.txt pymoose/requirements.in
-	pip install -r pymoose/requirements-dev.txt
+	CUSTOM_COMPILE_COMMAND="make pydep-upgrade" pip-compile --output-file=pymoose/requirements/base.txt pymoose/requirements/base.in
+	CUSTOM_COMPILE_COMMAND="make pydep-upgrade" pip-compile --output-file=pymoose/requirements/dev.txt pymoose/requirements/dev.in
+	pip install -r pymoose/requirements/base.txt
+	pip install -r pymoose/requirements/dev.txt
```

Comment:
This breakdown of the dependency is great. 
But I just noticed `requirements-dev.txt` is sill used in few places: under pydep in Makefile, Readme and in ci.yml

Reply:
good call -- fixed in latest commit